pkg/rsnotify/listeners/postgrespgx/pgxlistener.go

```diff
@@ -89,6 +90,15 @@ func NewPgxListener(args PgxListenerArgs) *PgxListener {
 }
 
 func (l *PgxListener) IP() string {
+	if l.ip != "" {
+		return l.ip
+	}
+
+	l.ip = l.ipReporter.IP()
+	go func() {
+		time.Sleep(5 * time.Minute)
+		l.ip = ""
```

Comment:
Do we need a RWMutex for clearing this value (and reading it)?

Reply:
Yeah, I wavered on this, but you could probably get into the situation with an empty string being returned. The "right," way is to probably setup a goroutine with a ticker and an RWMutex.

I'll take a look, I didn't want to get into making a stopper channel.